### PR TITLE
Tutorial step-03 with simplex support.

### PR DESCRIPTION
**Pull request by eliasstudiert**

### CODE REVIEW

File: tests/simplex/step-03.cc [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+  solution.reinit(dof_handler.n_dofs());
+  system_rhs.reinit(dof_handler.n_dofs());
+}
+
+template <int dim>
+void
+Step3<dim>::assemble_system()
+{
+  Simplex::QGauss<dim> quadrature_formula(dim == 2 ? (fe_degree == 1 ? 3 : 7) :
+                                                     (fe_degree == 1 ? 4 : 10));
```
Comment (peterrum) [inline tests/simplex/step-03.cc:160]:
```suggestion
  Simplex::QGauss<dim> quadrature_formula(fe_degree + 1);
```

File: tests/simplex/step-03.cc [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+int
+main()
+{
+  initlog(); 
+  
+  deallog.depth_console(2);
+
+  // refine mesh before conversion to simplex mesh
+  // Step3<2> laplace_problem_1;
+  // laplace_problem_1.run_1();
```
Comment (peterrum) [inline tests/simplex/step-03.cc:289]: ?

File: tests/simplex/step-03.cc [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+  void
+  setup_system();
+  void
+  assemble_system();
+  void
+  solve();
+  void
+  output_results() const;
+
+  Triangulation<dim, dim> in_triangulation, triangulation;
```
Comment (peterrum) [inline tests/simplex/step-03.cc:85]: `in_triangulation` is only temporal. You can remove it.

File: tests/simplex/step-03.cc [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+  unsigned int            fe_degree;
+  Simplex::FE_P<dim>      fe;
+  Simplex::FE_P<dim>      fe_mapping;
```
Comment (peterrum) [inline tests/simplex/step-03.cc:88]: Not needed. You can pass it to the mapping directly in the costructor.

File: tests/simplex/step-03.cc (hunk 1 above)
Comment (peterrum) [reply tests/simplex/step-03.cc:160]: Furthermore, could you make the quadrature formula a class field. Thanks.

File: tests/simplex/step-03.cc [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+
+  FullMatrix<double> cell_matrix(dofs_per_cell, dofs_per_cell);
+  Vector<double>     cell_rhs(dofs_per_cell);
+
+  std::vector<types::global_dof_index> local_dof_indices(dofs_per_cell);
+
+  for (const auto &cell : dof_handler.active_cell_iterators())
+    {
+      // re-compute values &gradients of shape func and determinant at quad -
+      // points
```
Comment (peterrum) [inline tests/simplex/step-03.cc:177]: Comments not needed.

File: tests/simplex/step-03.cc [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE.md at
+ * the top level directory of deal.II.
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * Authors: Wolfgang Bangerth, 1999,
+ *          Guido Kanschat, 2011
+ */
```
Comment (peterrum) [inline tests/simplex/step-03.cc:19]: Could you have a look at the header of https://github.com/dealii/dealii/blob/622107e2928bbe96110393da1c5bd986ba9b707b/tests/simplex/step-02.cc#L1-L22. In particular, could you make some notes if you had some issues while porting the tutorial.

File: tests/simplex/step-03.cc [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+  data_out.attach_dof_handler(dof_handler);
+  data_out.add_data_vector(solution, "solution");
+  // transfer data from the front- to the backend, transform into intermediate
+  // format
+  data_out.build_patches(mapping);
+
+  // open file and write data
+  // std::ofstream output("solution.vtk");
+  // data_out.write_vtk(output);
+  data_out.write_vtk(deallog.get_file_stream());
```
Comment (peterrum) [inline tests/simplex/step-03.cc:254]: The output is huge. Either you reduce the number of refinements or don't output the vtk file at all. Is there some relevant information printed to deallog?

File: tests/simplex/step-03.cc [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+void
+Step3<dim>::make_grid_1()
+{
+  unsigned int n_refinements = 3;
+  GridGenerator::subdivided_hyper_cube(in_triangulation,
+                                       2 * std::pow(2, n_refinements));
+  GridGenerator::convert_hypercube_to_simplex_mesh<dim, dim>(in_triangulation,
+                                                             triangulation);
+
+  std::cout << "Number of active cells: " << triangulation.n_active_cells()
```
Comment (peterrum) [inline tests/simplex/step-03.cc:119]: Don't print to std::cout but to deallog.

**Review by peterrum — COMMENTED** [review]
Some comments. These comments also apply to #11484.

File: tests/simplex/step-03.cc (hunk 2 above)
Comment (eliasstudiert) [reply tests/simplex/step-03.cc:289]: so we convert to tet first and then refine

File: tests/simplex/step-03.cc (hunk 5 above)
Comment (eliasstudiert) [reply tests/simplex/step-03.cc:177]: holds only for the comments in lines 176 & 177, or for all comments?

**Review by peterrum — APPROVED** [review]
@eliasstudiert I have some minor adjustments. Thanks for the contribution! This PR is ready from my side.

**Pull request merged**